src/client/datascience/serviceRegistry.ts

```diff
@@ -231,6 +232,9 @@ export function registerTypes(serviceManager: IServiceManager, inNotebookApiExpe
     serviceManager.addSingleton<IExtensionSingleActivationService>(IExtensionSingleActivationService, NotebookUsageTracker);
     serviceManager.addSingleton<IExtensionSingleActivationService>(IExtensionSingleActivationService, MigrateJupyterInterpreterStateService);
     serviceManager.addSingleton<IExtensionSingleActivationService>(IExtensionSingleActivationService, VariableViewActivationService);
+    if (isDevMode) {
+        serviceManager.addSingleton<IExtensionSingleActivationService>(IExtensionSingleActivationService, LogReplayService);

```

Comment:
Would it be better to add `IsDevMode` to the DI container as a constant.
Then in `activate` you can check the value and just exit instead of adding the command handlers?
That way we won't have to pass this around in future blocks of code.

Reply:
I had that originally but thought hey I could skip the entire class being loaded. Then it doesn't slow down activate at all.